cmd/editorconfig-checker/main_test.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"os"
+	"runtime"
+	"testing"
+)
+
+var mainHasRun chan int
+
+func captureReturncode(code int) {
+	mainHasRun <- code
+	runtime.Goexit()
+}
+
+func setArguments(args ...string) {
+	// os.Args needs to have the binary
+	os.Args = append([]string{"editorconfig-checker"}, args...)
+}
```

Comment:
This is not a good way to do.

I would suggest this


```suggestion
func setArguments(t *testing.T, args ...string) {
	t.Helper()

	initialArgs := os.Args

	// os.Args needs to have the binary
	os.Args = append([]string{"editorconfig-checker"}, args...)
	t.Cleanup(func() {
		os.Args = initialArgs
	})
}
```

Reply:
Would you just call it bad style to not restore the original arguments, or do you have any practical concerns? The only other consumer of `os.Args` is the `testing` package itself, but it has completed it's `os.Args` parsing by then.

Reply:
I reworked it according to your suggestions